### ISSUE TITLE
write_gml does not allow for empty lists as attributes

### ISSUE THREAD

**Issue by gtonkinhill** [networkx/networkx#4953]
The write_gml function does not handle empty lists as attributes. I'm not sure if this is expected behaviour.

### Current Behavior
Fails to convert an empty list to a string and raises the error `networkx.exception.NetworkXError: [] is not a string`

### Expected Behavior
I would expect an empty list to be written to the GML file as '[]'

### Steps to Reproduce
```
import networkx as nx
G = nx.complete_graph(4)
G.nodes[0]['test']=[]
nx.write_gml(G,'test.gml')
```

### Environment
Python version: 3.9.5
NetworkX version: 2.5.1

**Comment by rossbar** [comment]
I'm not sure what the "correct" behavior is either - what does the gml standard say about this case?

From the traceback, the problem lives in the default `stringizer`, specifically around: https://github.com/networkx/networkx/blob/dfa1150a9b8cf15183d7c910ff9b67ef3d220513/networkx/readwrite/gml.py#L729

This is what prevents empty lists being used.

Looking around the module a bit more, there is a function called `literal_stringizer` that seems to behave more the way you were expecting, try:

```python
nx.write_gml(G, 'test.gml', stringizer=nx.readwrite.gml.literal_stringizer)
```

**Comment by gtonkinhill** [comment]
Thanks very much for pointing out the alternative stringizer function. 
From what I can tell the GML standard does not specify what should be done in this case.
Incidentally, the link to the GML standard on the Networkx website appears to be out of date and now redirects to an alumnus page: http://www.infosun.fim.uni-passau.de/Graphlet/GML/gml-tr.html

**Comment by rossbar** [comment]
> Incidentally, the link to the GML standard on the Networkx website appears to be out of date and now redirects to an alumnus page: 

This should be fixed (insofar as is possible, as there doesn't seem to be an html version of the spec) in the [latest version of the documentation](https://networkx.org/documentation/latest/reference/readwrite/gml.html)

**Comment by dschult** [comment]
So, can we make this issue into a request to 

- fix the link to the GML standard
- add the literal_stringizer to the "See Also" section of the doc string.
- maybe include `literal_stringizer` in the main namespace.
  
Anything else?

**Comment by rossbar** [comment]
> So, can we make this issue into a request to
> 
>     * fix the link to the GML standard
This has been fixed in #4864 , though if anyone knows of an active html link (the current link is an archived page from wayback machine that only provides a pdf-version of the GML spec) this could certainly be improved.
>     * add the literal_stringizer to the "See Also" section of the doc string.
+1 for this.
>     * maybe include `literal_stringizer` in the main namespace.
I'm not sure about this one, since it is pretty specialized functionality that only really makes sense in the context of the GML module.
> 
> Anything else?

Makes sense to me - I'd say adding `literal_stringizer` to the `See Also` section of the `read_gml` docstring would be sufficient to close this.

**Comment by dschult** [comment]
I agree!  Thanks @rossbar 
Adding `literal_stringizer` to the See Also section should be sufficient.  No need to add it to the main namespace.
